pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.tsx

```diff
@@ -507,18 +534,18 @@ function BackgroundSelectDialog({ hide, current, select }: {
 
 	const availableBackgrounds = useMemo(() => assetManager.getBackgrounds(), [assetManager]);
 	const [nameFilter, setNameFilter] = useState('');
-	/*
-	 * TODO: Add a tag based filter to the dialog in a later version
-	 * const [tagFilter, setTagFilter] = useState('');
-	 */
+	const [tags, setTags] = useState<readonly string[]>([...assetManager.backgroundTags.keys()]);
+	const knownTagCategories = useMemo(() => uniq([...assetManager.backgroundTags.values()].map((tag) => tag.category)), [assetManager.backgroundTags]);
```

Comment:
Combining logic:
Having the tags combine with "OR" logic across categories is weird, especially with the fact that everything is selected by default - this basically means that one has to first deselect everything in all categories and then select the one thing one is searching for.
I think the categories should combine with an "AND" - categories where nothing is selected (which would be default) are ignored, items within category are combined with "OR" and categories themselves are combined with "AND".
Example selection:
- Time: Night
- Location: Bar, Beach
- Ambience, Item, Space: [nothing]

This selects all backgrounds that have "Night" tag _and_ have at least one of ["Bar", "Beach"] tags.